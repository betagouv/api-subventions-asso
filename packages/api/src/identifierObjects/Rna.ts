export const RNA_NAME = "rna";

export default class Rna {
    private rna: string;

    constructor(rna: string) {
        if (!Rna.isRna(rna)) {
            throw new Error("Invalid RNA: " + rna);
        }

        this.rna = rna;
    }

    static isRna(rna: string | undefined): boolean {
        return typeof rna === "string" && /^W\d[A-Z\d]\d{7}$/.test(rna);
    }

    static getName(): "rna" {
        return RNA_NAME;
    }

    get name(): "rna" {
        return RNA_NAME;
    }

    get value() {
        return this.rna;
    }

    toString() {
        return this.value;
    }

    equals(other: Rna) {
        return other.value === this.value;
    }
}
