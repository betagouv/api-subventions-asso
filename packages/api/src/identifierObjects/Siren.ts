import { IdentifierObject } from "./IdentifierObject";
import Siret from "./Siret";

export const SIREN_NAME = "siren";

export default class Siren extends IdentifierObject {
    static isSiren(siren: string | undefined): boolean {
        return typeof siren === "string" && /^\d{9}$/.test(siren);
    }

    static getName(): typeof SIREN_NAME {
        return SIREN_NAME;
    }

    // TODO: why "partialSiret" and not "siret" ?
    static fromPartialSiretStr(siret: string) {
        return new Siren(siret.slice(0, 9));
    }

    constructor(siren: string) {
        if (!Siren.isSiren(siren)) {
            throw new Error("Invalid Siren: " + siren);
        }

        super(siren);
    }

    get name(): typeof SIREN_NAME {
        return SIREN_NAME;
    }

    toSiret(nic: string) {
        return new Siret(this.identifier + nic);
    }
}
