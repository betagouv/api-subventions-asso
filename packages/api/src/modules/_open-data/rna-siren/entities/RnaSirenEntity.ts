import { Rna, Siren } from "dto";

export default class RnaSiren {
    constructor(public rna: Rna, public siren: Siren, public names?: string[]) {}
}
