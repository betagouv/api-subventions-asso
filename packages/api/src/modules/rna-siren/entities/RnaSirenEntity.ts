import { Rna, Siren } from "../../../@types";

export default class RnaSiren {
    constructor(
        public rna: Rna,
        public siren: Siren,
    ) {}
}