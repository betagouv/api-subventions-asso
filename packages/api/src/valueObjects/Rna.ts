export default class Rna {
    private rna: string;

    constructor(rna: string) {
        if (!Rna.isRna(rna)) {
            throw new Error("Invalid RNA: " + rna);
        }

        this.rna = rna;
    }

    static isRna(rna: string | undefined): boolean {
        return typeof rna === "string" && /^W\d[A-Z\d]\d{7}$/.test(rna);
    }

    get name() {
        return "rna";
    }

    get value() {
        return this.rna;
    }

    toString() {
        return this.value;
    }

    equals(other: Rna) {
        return other.value === this.value;
    }
}
