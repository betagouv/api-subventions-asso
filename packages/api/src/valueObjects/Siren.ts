import Siret from "./Siret";

export default class Siren {
    private siren: string;

    constructor(siren: string) {
        if (!Siren.isSiren(siren)) {
            throw new Error("Invalid Siren: " + siren);
        }

        this.siren = siren;
    }

    static isSiren(siren: string | undefined): boolean {
        return typeof siren === "string" && /^\d{9}$/.test(siren);
    }

    get value() {
        return this.siren;
    }

    toString() {
        return this.value;
    }

    equals(other: Siren) {
        return other.value === this.value;
    }

    toSiret(nic: string) {
        return new Siret(this.siren + nic);
    }
}
