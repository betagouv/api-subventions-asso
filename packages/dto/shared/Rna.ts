/**
 * Identifiant Rna
 * @pattern ^W\d[A-Z\d]\d{7}$
 */
export type RnaDto = string;
