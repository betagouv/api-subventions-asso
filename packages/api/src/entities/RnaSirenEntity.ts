import Rna from "../valueObjects/Rna";
import Siren from "../valueObjects/Siren";

export default class RnaSirenEntity {
    constructor(public rna: Rna, public siren: Siren, public id?: string) {}
}
