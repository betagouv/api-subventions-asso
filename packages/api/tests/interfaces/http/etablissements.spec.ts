import request from "supertest";
import { createAndGetAdminToken, createAndGetUserToken } from "../../__helpers__/tokenHelper";
import associationsService from "../../../src/modules/associations/associations.service";
import { BadRequestError } from "core";
import DEFAULT_ASSOCIATION, {
    API_ASSO_ESTABLISHMENTS_FROM_SIREN,
    LONELY_RNA,
    SIREN_STR,
    SIRET_STR,
} from "../../__fixtures__/association.fixture";
import rnaSirenPort from "../../../src/dataProviders/db/rnaSiren/rnaSiren.port";
import Rna from "../../../src/identifierObjects/Rna";
import RnaSirenEntity from "../../../src/entities/RnaSirenEntity";
import Siren from "../../../src/identifierObjects/Siren";
import statsAssociationsVisitPort from "../../../src/dataProviders/db/stats/statsAssociationsVisit.port";
import { App } from "supertest/types";
import apiAssoService from "../../../src/modules/providers/apiAsso/apiAsso.service";
import applicationFlatPort from "../../../src/dataProviders/db/applicationFlat/applicationFlat.port";
import paymentFlatPort from "../../../src/dataProviders/db/paymentFlat/paymentFlat.port";
import {
    APPLICATION_LINK_TO_CHORUS,
    APPLICATION_LINK_TO_FONJEP,
} from "../../../src/modules/applicationFlat/__fixtures__";
import {
    CHORUS_PAYMENT_FLAT_ENTITY,
    FONJEP_PAYMENT_FLAT_ENTITY,
    FONJEP_PAYMENT_FLAT_ENTITY_2,
} from "../../../src/modules/paymentFlat/__fixtures__/paymentFlatEntity.fixture";

const g = global as unknown as { app: App };

const ETABLISSEMENT_SIRET = SIRET_STR;

async function initData() {
    await applicationFlatPort.insertMany([APPLICATION_LINK_TO_CHORUS, APPLICATION_LINK_TO_FONJEP]);

    // PAYMENT FLAT
    await paymentFlatPort.insertMany([
        CHORUS_PAYMENT_FLAT_ENTITY,
        FONJEP_PAYMENT_FLAT_ENTITY,
        FONJEP_PAYMENT_FLAT_ENTITY_2,
    ]);
}

describe("/etablissement", () => {
    beforeAll(async () => {
        jest.spyOn(apiAssoService, "findEstablishmentsBySiren").mockResolvedValue(API_ASSO_ESTABLISHMENTS_FROM_SIREN);
    });

    beforeEach(async () => {
        await rnaSirenPort.insert(
            new RnaSirenEntity(new Rna(DEFAULT_ASSOCIATION.rna), new Siren(DEFAULT_ASSOCIATION.siren)),
        );
        await initData();
    });

    describe("/siret/subventions", () => {
        it("returns subventions", async () => {
            const response = await request(g.app)
                .get(`/etablissement/${ETABLISSEMENT_SIRET}/subventions`)
                .set("x-access-token", await createAndGetUserToken())
                .set("Accept", "application/json");

            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
    });

    describe("/siret/grants", () => {
        it("returns grants", async () => {
            const response = await request(g.app)
                .get(`/etablissement/${ETABLISSEMENT_SIRET}/grants`)
                .set("x-access-token", await createAndGetUserToken())
                .set("Accept", "application/json");
            expect(response.statusCode).toBe(200);
            expect(response.body).toMatchSnapshot();
        });
    });

    describe("/siret", () => {
        it("should add one visits on stats AssociationsVisit", async () => {
            const beforeRequestTime = new Date();
            await rnaSirenPort.insert(new RnaSirenEntity(new Rna(LONELY_RNA), new Siren(SIREN_STR)));
            await request(g.app)
                .get(`/etablissement/${SIRET_STR}`)
                .set("x-access-token", await createAndGetUserToken())
                .set("Accept", "application/json");

            const actual = (await statsAssociationsVisitPort.findOnPeriod(beforeRequestTime, new Date()))[0];
            expect(actual).toMatchObject({
                associationIdentifier: SIRET_STR.slice(0, 9),
            });
        });

        it("should not add one visits on stats AssociationsVisit beacause user is admin", async () => {
            const beforeRequestTime = new Date();
            await request(g.app)
                .get(`/etablissement/${SIRET_STR}`)
                .set("x-access-token", await createAndGetAdminToken())
                .set("Accept", "application/json");

            const actual = await statsAssociationsVisitPort.findOnPeriod(beforeRequestTime, new Date());
            expect(actual).toHaveLength(0);
        });

        it("should not add one visits on stats AssociationsVisit beacause user is not authentified", async () => {
            const beforeRequestTime = new Date();
            await request(g.app).get(`/etablissement/${SIRET_STR}`).set("Accept", "application/json");

            const actual = await statsAssociationsVisitPort.findOnPeriod(beforeRequestTime, new Date());
            expect(actual).toHaveLength(0);
        });

        it("should not add one visits on stats AssociationsVisit beacause status is not 200", async () => {
            const beforeRequestTime = new Date();
            jest.spyOn(associationsService, "getAssociation").mockImplementationOnce(() => {
                throw new BadRequestError();
            });
            await request(g.app).get(`/etablissement/${SIRET_STR}`).set("Accept", "application/json");

            const actual = await statsAssociationsVisitPort.findOnPeriod(beforeRequestTime, new Date());
            expect(actual).toHaveLength(0);
        });
    });
});
