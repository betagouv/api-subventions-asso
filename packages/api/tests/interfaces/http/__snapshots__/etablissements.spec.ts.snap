// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/etablissement /siret/grants returns grants 1`] = `
{
  "subventions": [
    {
      "application": {
        "actions_proposee": [
          {
            "intitule": {
              "last_update": "2025-12-12T00:00:00.000Z",
              "provider": "osiris",
              "type": "string",
              "value": "objet",
            },
          },
        ],
        "annee_demande": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "number",
          "value": 2023,
        },
        "co_financement": {
          "cofinanceur": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "string",
            "value": "",
          },
          "cofinanceur_email": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "string",
            "value": "",
          },
          "montants": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "number",
            "value": 0,
          },
        },
        "creer_le": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "object",
          "value": "2023-03-12T00:00:00.000Z",
        },
        "date_commision": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "object",
          "value": "2023-03-13T00:00:00.000Z",
        },
        "dispositif": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "dispositif",
        },
        "ej": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "EJ0001",
        },
        "financeur_principal": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "attribuant",
        },
        "montants": {
          "accorde": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "number",
            "value": 15000,
          },
          "demande": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "number",
            "value": 30000,
          },
          "total": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "number",
            "value": 50000,
          },
        },
        "pluriannualite": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "boolean",
          "value": true,
        },
        "service_instructeur": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "serviceInstructeur",
        },
        "siret": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "10000000000001",
        },
        "sous_dispositif": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "sous-dispositif",
        },
        "status": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "Accordé",
        },
        "statut_label": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "Accordé",
        },
        "transmis_le": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "object",
          "value": "2023-03-12T00:00:00.000Z",
        },
        "versementKey": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "1000000000019--subv001--2023",
        },
      },
      "payments": [
        {
          "activitee": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "Label d'activité Exemple",
          },
          "amount": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "number",
            "value": 89988.3,
          },
          "centreFinancier": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "UO DGER XXXX-C001",
          },
          "dateOperation": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "object",
            "value": "2023-07-12T00:00:00.000Z",
          },
          "domaineFonctionnel": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "Label d'action Exemple",
          },
          "ej": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "0001821732",
          },
          "libelleProgramme": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "Programme Exemple",
          },
          "programme": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "number",
            "value": 101,
          },
          "siret": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "10000000000001",
          },
          "versementKey": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "chorus",
            "type": "string",
            "value": "1000000000019--subv001--2023",
          },
        },
      ],
    },
    {
      "application": {
        "actions_proposee": [
          {
            "intitule": {
              "last_update": "2025-12-12T00:00:00.000Z",
              "provider": "fonjep",
              "type": "string",
              "value": "objet",
            },
          },
        ],
        "annee_demande": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "number",
          "value": 2023,
        },
        "co_financement": {
          "cofinanceur": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "",
          },
          "cofinanceur_email": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "",
          },
          "montants": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 0,
          },
        },
        "creer_le": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "object",
          "value": "2023-03-12T00:00:00.000Z",
        },
        "date_commision": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "object",
          "value": "2023-03-13T00:00:00.000Z",
        },
        "dispositif": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "dispositif",
        },
        "ej": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "N/A",
        },
        "financeur_principal": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "attribuant",
        },
        "montants": {
          "accorde": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 15000,
          },
          "demande": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 30000,
          },
          "total": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 50000,
          },
        },
        "pluriannualite": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "boolean",
          "value": true,
        },
        "service_instructeur": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "serviceInstructeur",
        },
        "siret": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "10000000000001",
        },
        "sous_dispositif": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "sous-dispositif",
        },
        "status": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "Accordé",
        },
        "statut_label": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "Accordé",
        },
        "transmis_le": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "object",
          "value": "2023-03-12T00:00:00.000Z",
        },
        "versementKey": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "S03351-2023-05-12-2023-10000000000001",
        },
      },
      "payments": [
        {
          "amount": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 3752,
          },
          "codePoste": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "S03351",
          },
          "dateOperation": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "object",
            "value": "2023-07-12T00:00:00.000Z",
          },
          "libelleProgramme": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "Programme Exemple",
          },
          "programme": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 163,
          },
          "siret": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "10000000000001",
          },
          "versementKey": {
            "last_update": "2023-07-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "S03351-2023-05-12-2023-10000000000001",
          },
        },
        {
          "amount": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 3752,
          },
          "codePoste": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "S03351",
          },
          "dateOperation": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "object",
            "value": "2023-11-12T00:00:00.000Z",
          },
          "libelleProgramme": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "Programme Exemple",
          },
          "programme": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "number",
            "value": 163,
          },
          "siret": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "10000000000001",
          },
          "versementKey": {
            "last_update": "2023-11-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "S03351-2023-05-12-2023-10000000000001",
          },
        },
      ],
    },
  ],
}
`;

exports[`/etablissement /siret/subventions returns subventions 1`] = `
{
  "subventions": [
    {
      "actions_proposee": [
        {
          "intitule": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "osiris",
            "type": "string",
            "value": "objet",
          },
        },
      ],
      "annee_demande": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "number",
        "value": 2023,
      },
      "co_financement": {
        "cofinanceur": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "",
        },
        "cofinanceur_email": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "string",
          "value": "",
        },
        "montants": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "number",
          "value": 0,
        },
      },
      "creer_le": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "object",
        "value": "2023-03-12T00:00:00.000Z",
      },
      "date_commision": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "object",
        "value": "2023-03-13T00:00:00.000Z",
      },
      "dispositif": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "dispositif",
      },
      "ej": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "EJ0001",
      },
      "financeur_principal": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "attribuant",
      },
      "montants": {
        "accorde": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "number",
          "value": 15000,
        },
        "demande": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "number",
          "value": 30000,
        },
        "total": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "osiris",
          "type": "number",
          "value": 50000,
        },
      },
      "pluriannualite": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "boolean",
        "value": true,
      },
      "service_instructeur": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "serviceInstructeur",
      },
      "siret": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "10000000000001",
      },
      "sous_dispositif": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "sous-dispositif",
      },
      "status": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "Accordé",
      },
      "statut_label": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "Accordé",
      },
      "transmis_le": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "object",
        "value": "2023-03-12T00:00:00.000Z",
      },
      "versementKey": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "osiris",
        "type": "string",
        "value": "1000000000019--subv001--2023",
      },
    },
    {
      "actions_proposee": [
        {
          "intitule": {
            "last_update": "2025-12-12T00:00:00.000Z",
            "provider": "fonjep",
            "type": "string",
            "value": "objet",
          },
        },
      ],
      "annee_demande": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "number",
        "value": 2023,
      },
      "co_financement": {
        "cofinanceur": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "",
        },
        "cofinanceur_email": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "string",
          "value": "",
        },
        "montants": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "number",
          "value": 0,
        },
      },
      "creer_le": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "object",
        "value": "2023-03-12T00:00:00.000Z",
      },
      "date_commision": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "object",
        "value": "2023-03-13T00:00:00.000Z",
      },
      "dispositif": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "dispositif",
      },
      "ej": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "N/A",
      },
      "financeur_principal": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "attribuant",
      },
      "montants": {
        "accorde": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "number",
          "value": 15000,
        },
        "demande": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "number",
          "value": 30000,
        },
        "total": {
          "last_update": "2025-12-12T00:00:00.000Z",
          "provider": "fonjep",
          "type": "number",
          "value": 50000,
        },
      },
      "pluriannualite": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "boolean",
        "value": true,
      },
      "service_instructeur": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "serviceInstructeur",
      },
      "siret": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "10000000000001",
      },
      "sous_dispositif": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "sous-dispositif",
      },
      "status": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "Accordé",
      },
      "statut_label": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "Accordé",
      },
      "transmis_le": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "object",
        "value": "2023-03-12T00:00:00.000Z",
      },
      "versementKey": {
        "last_update": "2025-12-12T00:00:00.000Z",
        "provider": "fonjep",
        "type": "string",
        "value": "S03351-2023-05-12-2023-10000000000001",
      },
    },
  ],
}
`;
